canvas-image-worker.js

```diff
@@ -1,18 +1,66 @@
 function draw (data) {
 
-	if (data.processing.greyscaleMethod == "luminance") {
+	if (data.processing.grayscaleMethod == "luminance") {
 
-		greyscale_luminance(data.image.data);
+		grayscale_luminance(data.image.data);
 
-	} else if (data.processing.greyscaleMethod == "average") {
+	} else if (data.processing.grayscaleMethod == "decomposition") {
 
-		greyscale_average(data.image.data);
+		grayscale_decomposition(data.image.data);
+
+	} else if (data.processing.grayscaleMethod == "desaturation") {
+
+		grayscale_desaturation(data.image.data);
+
+	} else if (data.processing.grayscaleMethod == "shades") {
+
+		grayscale_shades(data.image.data, data.processing.grayscaleShades);
+
+	} else if (data.processing.grayscaleMethod == "singlechannel") {
+
+		grayscale_singlechannel(data.image.data, data.processing.grayscaleChannel);
+
+	} else if (data.processing.grayscaleMethod == "average") {
+
+		grayscale_average(data.image.data);
 
 	}
 
 	if (data.processing.ditherMethod == "atkinson") {
 
-		dither_atkinson(data.image.data, data.image.width, (data.processing.greyscaleMethod == ""));
+		dither_atkinson(data.image.data, data.image.width, (data.processing.grayscaleMethod == ""));
+
+	} else if (data.processing.ditherMethod == "burkes") {
+
+		dither_burkes(data.image.data, data.image.width, (data.processing.grayscaleMethod == ""));
+
+	} else if (data.processing.ditherMethod == "falsefloydsteinberg") {
+
+		dither_falsefloydsteinberg(data.image.data, data.image.width, (data.processing.grayscaleMethod == ""));
+
+	} else if (data.processing.ditherMethod == "floydsteinberg") {
+
+		dither_floydsteinberg(data.image.data, data.image.width, (data.processing.grayscaleMethod == ""));
+
+	} else if (data.processing.ditherMethod == "jarvisjudiceninke") {
+
+		dither_jarvisjudiceninke(data.image.data, data.image.width, (data.processing.grayscaleMethod == ""));
+
+	} else if (data.processing.ditherMethod == "sierra") {
+
+		dither_sierra(data.image.data, data.image.width, (data.processing.grayscaleMethod == ""));
+
+	} else if (data.processing.ditherMethod == "sierratworow") {
+
+		dither_sierratworow(data.image.data, data.image.width, (data.processing.grayscaleMethod == ""));
+
+	} else if (data.processing.ditherMethod == "sierralite") {
+
+		dither_sierralite(data.image.data, data.image.width, (data.processing.grayscaleMethod == ""));
+
+	} else if (data.processing.ditherMethod == "stucki") {
+
+		dither_stucki(data.image.data, data.image.width, (data.processing.grayscaleMethod == ""));
```

Comment:
All the `data.processing.grayscaleMethod == ""` bits need to instead check for your newly-introduced `"disable"` value, otherwise the colour versions don’t work.

```suggestion
		dither_atkinson(data.image.data, data.image.width, (data.processing.grayscaleMethod == "disable"));

	} else if (data.processing.ditherMethod == "burkes") {

		dither_burkes(data.image.data, data.image.width, (data.processing.grayscaleMethod == "disable"));

	} else if (data.processing.ditherMethod == "falsefloydsteinberg") {

		dither_falsefloydsteinberg(data.image.data, data.image.width, (data.processing.grayscaleMethod == "disable"));

	} else if (data.processing.ditherMethod == "floydsteinberg") {

		dither_floydsteinberg(data.image.data, data.image.width, (data.processing.grayscaleMethod == "disable"));

	} else if (data.processing.ditherMethod == "jarvisjudiceninke") {

		dither_jarvisjudiceninke(data.image.data, data.image.width, (data.processing.grayscaleMethod == "disable"));

	} else if (data.processing.ditherMethod == "sierra") {

		dither_sierra(data.image.data, data.image.width, (data.processing.grayscaleMethod == "disable"));

	} else if (data.processing.ditherMethod == "sierratworow") {

		dither_sierratworow(data.image.data, data.image.width, (data.processing.grayscaleMethod == "disable"));

	} else if (data.processing.ditherMethod == "sierralite") {

		dither_sierralite(data.image.data, data.image.width, (data.processing.grayscaleMethod == "disable"));

	} else if (data.processing.ditherMethod == "stucki") {

		dither_stucki(data.image.data, data.image.width, (data.processing.grayscaleMethod == "disable"));
```

Alternately, the new value could be removed from the form.

Reply:
Also this bug was already fixed, nice spot!